### ISSUE TITLE
Add sortable and repeating group in Theme Options

### ISSUE THREAD

**Issue by padalec** [vafour/vafpress-framework#85]
Hi,
just like in topic I need sortable and repeating field in Theme Option. I try to do this by group section but this returns fatal error. Adding sortable or repeating parameter in section doesn't work either:
array(
            'type'      => 'section',  
            'title'     => 'some label',
            'name'      => 'name',
            'fields'    => array(....),
            'sortable'  => true,
            'repeating' => true,
        ); 
Best Regards

**Comment by vladan-me** [comment]
To me it looks like it simply isn't supported for theme options, only for metaboxes. Take a look at file classes/metabox.php and compare it to classes/option.php
Simple search through files gets to point where there's nothing about string 'sortable' inside option.php

**Comment by padalec** [comment]
Yes I know there isn't supported in Theme Options but this functionality can be very useful

**Comment by vladan-me** [comment]
Yes, I see. I can't see authors actually replying for a long time on this project, might be that they are busy with other stuff... I have few more requests (and solutions) as well but unsure if should start a fork or just wait for authors to make some reply on current status of this project.

**Comment by padalec** [comment]
Ehh I have the same problem some of replay for my question I waiting from almost month maybe they are in holidays:)

**Comment by thednp** [comment]
@vladan-me did you manage to make it work as you described?

**Comment by vladan-me** [comment]
No, sorry, I didn't need this particular feature but I did made some customizations to those files (for features I need) and I tried to understand structure of this framework. Also, I'd like to add that I've sent an email to authors of this framework about current status and got no reply at all. :(

**Comment by thednp** [comment]
@vladan-me FYI: I've had talks with devs on new features and so on, they're all excited with community inputs and so on, but nobody ever supported the project with donations or purchasing of the devs works.

What they had on tables some time ago was: 
- start making PRO, paid, extensions for the framework such as powerful fields, page builder fields, better multimedia browsing, etc
- or give the framework ownership to someone else.

As far as I can see, none of the above happened.

**Comment by vladan-me** [comment]
Interesting, I was assuming that something is going on. I don't know, I think it is big self promotion to have framework like this and in order to keep that status it's logical to continue active development. Anyway, if it means something, I'd probably purchase paid version of this framework with all those additional features.

**Comment by scrobbleme** [comment]
I could need this feature as well. As of now, I implemented my feature as custom post type to be able to use metaboxes.

It would be really sad to see this framework vanish, but I don't understand their point. The do not have any premium themes or plugins for sale on their website. So what should people buy? 
Also they could sell a whitelabel version or just a version without GPL license.

Maybe they just could start incorporating pull requests.

**Comment by padalec** [comment]
I agree with you. Maybe they should sell some premium support or just like WPML should by payed. But I thing the GPL license is required if you want sell theme on themeforest. Someone from vafpress should say something about what they plan do with this framework.

**Comment by vladan-me** [comment]
Well, long time no answer from authors... I've started a [fork](https://github.com/vladan-me/vafpress-framework/tree/develop) and all contributions are welcome. :) I invite all of you to help making this perspective framework even better.
Currently will try to develop in "develop" branch. I've added few of fixes that are (somewhat) critical.

**Comment by wpsoul** [comment]
I think that paid pro version is good solution. I can pay for this and I think many peoples also can do this. My needs that I can donate:
1. Ability to use metafields in taxonomy
2. Custom sidebar area creating
3. Ability to use some fields in widget (color, multiselect, image checkbox, radiobtn)
4. Speed improvements

**Comment by padalec** [comment]
@vladan-me maybe this is a good idea but we don't have any documentation for this framework so fixes and new issues can by very hard too implemented.

**Comment by scrobbleme** [comment]
@vladan-me Looks like a good way to start over ;)

**Comment by ghost** [comment]
Can I ask why you guys bother to use VafPress since it is hardly supported? Why not use Redux which has great support, is updated almost weekly, is easy to use, is a premium product in itself, yet free.
https://github.com/ReduxFramework/redux-framework

**Comment by scrobbleme** [comment]
@cmwwebfx The reasons I stared using Vafpress was it's simplicity, clean documentation and good set of features. 
Also, you see, that there is some community support here.

**Comment by ghost** [comment]
Ahh right. I saw this morning someone is forking it because things are not working and no support is given or around on this project.

**Comment by wpsoul** [comment]
@cmwwebfx Because redux - is just theme option panel. Vafpress is theme option + shortcode generator + meta fields constructor.

**Comment by ghost** [comment]
Ahh ok I understand better what you need then. If you need framework + short code generator + meta boxes all in one solution, then you can use Titan framework, or alternatively you can do this in caldera engine pro. Both have all the above.

**Comment by dovy** [comment]
Lead dev of Redux here. We have metaboxes, we just done have short codes yet. That would be really easy to do. 

We have widget areas too. ;)

**Comment by wpsoul** [comment]
@cmwwebfx I know about Titan, but it doesn't have a repeatable and sortable groups in meta fields. Also, metafields in Vafpress can be assigned to page template. It's very useful for me.

@dovy I like Redux Framework, sometimes I think to move my themes to Redux, but I don't like money policy of Redux. Metaboxes + Shortcode + icon chooser = near $200. I think metaboxes for 125$ - is high price for one extension. For example, ACF offers developer license with all pro extension for $100. Vafpress offers very similar functions, but for free. If you reduce price for metaboxes extension, I think I will move my commercial products to your framework. 100$ for pack of these 3 extensions will be good for me :-)

**Comment by vladan-me** [comment]
@dovy Yes, you do have, but difference is that this is open source and has decent functionality already and you are charging for MetaBox $125/year + Shortcodes $30/year + Widget Areas $90/year :)

**Comment by wpsoul** [comment]
The only framework that is also open sourced and has such functions - Piklist, but they are still in beta and I don't like the way of implementing Piklist to theme. Maybe it's good for creating sites, but bad for creating themes and different complex solutions

**Comment by ghost** [comment]
Piklist is ok, but certainly would not use it building a theme. No way.. 
If you want something that is incredible, then hopefully Redux will either make a bundle to be competitive in  pricing that even the starter developer will buy into it. Alternatively Caldera Engine will do much more than you think, and even repeatable grouped fields. And when you are done you can simply export out the code to ready embed in a plugin or theme without the dependency of needing to install Caldera in a project. This is the biggest downfall of Piklist TBH.

Screenshots here, and keep in mind Caldera Engine is built by one of the devs that built Pods.

https://www.dropbox.com/s/xi69n4kxjbmjlu9/Screenshot%202014-09-12%2021.33.03.png?dl=0

https://www.dropbox.com/s/lk2mhspxfjbcjy1/Screenshot%202014-09-12%2021.34.14.png?dl=0

https://www.dropbox.com/s/hcxsz830jxa4h9e/Screenshot%202014-09-12%2021.36.10.png?dl=0

https://www.dropbox.com/s/ivz9k9nz6umfv1q/Screenshot%202014-09-12%2021.37.08.png?dl=0

**Comment by ghost** [comment]
And costs only $50.

**Comment by ghost** [comment]
From what I hear, it might be possible something big is coming in the way of bundles from Redux. This will open the gates to those developers that are starting to get into developing themes and plugins. Hopefully @dovy can advise further on this at some stage when this might happen.

**Comment by padalec** [comment]
@cmwwebfx to be honest I do not prefer coding like in this framework (create Post Type by clicking mouse buttons) :). If you have good OO structure you can create what you want very easy an fast. I very like Vfpress framework because in it there is OO structure and have everything what I need to create theme - Short Code generator, Theme Options and Meta Boxes.

**Comment by ghost** [comment]
It is no different to using GenerateWP other than it is a whole lot more powerful. If you think Caldera Engine is point and click, then you are terribly mistaken. You still have to code.

**Comment by ghost** [comment]
Also, the work you do can only be as good as the sharpest tool in your toolbox ;)

**Comment by padalec** [comment]
Maybe you have right I never used Caldera Engine :) but I have me own based on Vafpress :)

**Comment by ghost** [comment]
I don't disagree at all, Vafpress is lovely, and really slick and feature rich. But the issue is the support on it seems extinct like it has 3am Herpes. 
I would love to see this pick up and be developed further. I would rather pay for something that is well supported. Free is nice, but comes with a price.

**Comment by padalec** [comment]
Yes you have right about the support :( I think the only way to pick up this framework we(people using vafpress) should to develop.

**Comment by dovy** [comment]
Hi all, I'd like to give you better perspective of where Redux is at.

Redux 3.0 started by merging 2 older frameworks that lost support: SMOF and NHP. We believed that if we could rally all the devs together and for our efforts they'd donate and support the Redux project. Unfortunately like many projects, donations were few and very far between.

If you look at Redux, it's a bit different than VafPress. Our core is fully open source, and the community drives all development. We're extremely active on our issue tracker. Right now the issues are at 6 open, 1487 closed... What's even more impressive is 90% of that support comes from mainly 2 people, @kprovance and myself.

Our (known) developers are very successful. http://reduxframework.com/showcase/envato   Over $6.2+ million in gross sales this year. In saying that, over the last year we've had 0.032% of that through donations... The few extension sales that we have had has helped in the developemnt of the project.

We soon realized that if we were to keep things progressing we'd have to sell something at a premium. We didn't want to make Redux proprietary. So, we started selling extensions. Keeping this amazing core free for all while recieving some compensation to keep the developement alive.

We ensure the extensions are of the same high quality as the Redux core, and are deserving of their set price. Most of the developers that we talk to are wanting to pay something that has premium support and maintained at a high level. In life you pay for what you get. 

We're totally willing to reduce the price provided we don't reduce revenue, because it's pretty low right now as it is. If you have a better model suggestion, then we are 100% all ears, and want to listen. 

As @cmwwebfx mentioned, we'll probably be offering a bundle here to try to reduce the barrier to entry.

So to recap:
- Most powerful framework out there - FREE
- Extending that framework in powerful ways - PREMIUM (or extend it yourself)

Our goal is simply to keep Redux alive. If we can do that, we can keep making amazing stuff. Please feel free to pass any ideas you may have on a sustainable model or product bundle pricing. We'd love to hear your thoughts.

Let's discuss this openly.

**Comment by dovy** [comment]
Oh and I'm about 70% of the way through the Vafpress to Redux converter.  :)

**Comment by vladan-me** [comment]
Ok, let's put my view on the paper. 
First, what matters to me the most is what matters potential users of our theme and that is obviously, to put in simple words:
1) look
2) feel (features)
3) speed

Second, what I'd like to see the most in framework is
4) code base
5) support
6) documentation
7) extensibility

Now let's make comparison:

Vafpress framework
1) slick, eye-catching
2) can make most of it, few important features missing
3) works quite well by now
4) is based on good concept, has some flaws though
5) support is lacking completely at this point
6) documentation is excellent, covers everything required + additional examples in form of files that can be used as is.
7) you are able to extend some of classes, not all effectively though

Redux framework
1) although is not very different, there are some visual flaws that just makes you feel, "this is not it", e.g. font sizes not always appropriate, margins between fields, there's just some bit missing, some designers eye will see this even better than me
2) didn't try honestly but from what I see from demo & material it looks like it has everything that users need and what is more important, you are open for new (sane) requests
3) checking at php code base and bit of javascript, seems that will have problems with speed of execution which is confirmed by some of reports in your issue tracker
4) it's decent but feels over-engineered, I don't know, I can compare it to Avada's theme options, you get tired while you see all those "features", I personally don't like that concept
5) support is great and actively trying to resolve issues
6) documentation is extensive but, hm, "cranky", feels like you are missing good designer (related to 1) as well)
7) N/A, can't really talk about this without using it for a while

**Comment by vladan-me** [comment]
@dovy First, I really admire your dedication in working on Redux. 
While I support your view on selling premium extensions I feel sometimes you really went over limits of reasonable. It could be life in USA is very expensive :) When there's [free plugin](https://wordpress.org/plugins/custom-sidebars/screenshots/) that does that and even more, selling Widget Areas as extension plugin for a price of $90/year, I am sorry but it is just arrogant.

**Comment by kprovance** [comment]
<blockquote>
I am sorry but it is just arrogant.
</blockquote>

You know what they say about opinions, yeah?

**Comment by vladan-me** [comment]
@kprovance Yup, is like ..., everyone has one :)

**Comment by vladan-me** [comment]
@padalec Let's try and see how it goes, I feel that I understand this framework enough to make some extensions but still need a lot of support. We can split some tasks. Anyway, my skype id is vladan.. (with two dots) so from Monday best, to have talk and see what can be done about it.

**Comment by ghost** [comment]
@vladan-me I agree 100% on what you mention about "some" things not looking great or lining up. It is one of my pet hates when things are out of line. For reference on this thread on what doesn't quite fit your ideas of the perfect options panel, I would love to see what your ideas are on what doesn't fit the bill.

Many things seems to have changed and updated almost weekly, and with those updates many little things that were not looking right for me seems to have been adjusted. 

As for the Custom Sidebars plugin https://wordpress.org/plugins/custom-sidebars/, I have been using that awhile now and does an outstanding job for a free plugin, and agree that paying $90 a year is too much.

What do you @wpsoul, @thednp  and @padalec think the pricing packaging should be to be competitive on the market, yet be affordable for you and me that are not making huge $$ like some of the developers out there. I will be interested to see what @dovy might come up with if our small voices or wishes are even considered.

**Comment by padalec** [comment]
@cmwwebfx for me is very good idea. But first we should collect some more contributors too develop this framework because  2-3 developers is not enough. Guys maybe we should connect on facebook chat so we can refine some details?

**Comment by scrobbleme** [comment]
I would like to help too. We have currently one premium plugin available, which uses Vafpress and are planning to release 2 more this year. Hopefully we can generate some income and give back.

**Comment by dovy** [comment]
Hey all, sorry for my absence. It was WordCamp SLC this past weekend, so I was a bit busy.

@wpsoul I appreciate your input about pricing. I would like to point out one thing though. ACF is almost a consumer level product. Standard users even use it. It's very easy and has a MASSIVE following. So a $30 price tag for an extension could easily mean a large return given the user base. Redux is very much developer/designer centric, so our audience is much smaller. I could say yea, we can do $30. But look at the other frameworks and extensions (like the [Titan Shortcode Extension](http://codecanyon.net/item/titan-framework-shortcode-extension/7009811?WT.ac=search_item&WT.oss_phrase=titan%20shortcode&WT.oss_rank=1&WT.z_author=GambitTech) on CodeCanyon, which also has a $135 price tag within products) and you can see pretty quickly why we set the price as we have. $125 amounts to 3 or 4 if total sales. Visual Composer is $140 for every theme you sell it with. I fail to see how this is any different.

@vladan-me I really appreciate your input. I'd love to respond to a few things if you wouldn't mind.

I had actually never seen that plugin. I based my pricing on what CodeCanyon uses for extended licensing for products, as I pointed out to @wpsoul. I assure you, there's no arrogance here. Just an attempt to keep our project alive.

Now onto your second message. I have to ask, have you looked at Redux lately (last 5 months)? We are using every native WordPress styling possible. We've even made the panel change to match the current admin style. Vafpress is, I believe, bootstrap. Also if you're basing your thoughts on the screenshots on the front page of our site, I'd ask you instead install the plugin. We need to update those screenshots and they're from 3rd quarter of last year. Things have changed dramatically.

Now on to each of your points of Redux.
1)  I'd love to know some examples. We've really tried to clean that up over the past 5 months. They were primarily legacy issue with the old SMOF css.
3) Very much in the past. Now we're actually quite speedy given our lazy-load javascript. Panels with hundreds of options are responding as fast as panels with a few.
4) It's s swiss army knife for sure. The thing is, you don't have to use all the options. It's just capable of any use case in WordPress you can think of.
6) I'm confused about this statement... [Redux Docs](http://docs.reduxframework.com) vs [VafPress Docs](http://vafpress.com/documentation/vafpress-framework/getting-started.html). I'm pretty sure ours has more of a designer touch. Can you elaborate what you mean? Maybe you were talking about the Wiki way back when? For example, this is the documentation for the [Select Field](http://docs.reduxframework.com/core/fields/select/). What do you dislike about it? I really would like to improve this even if you're not interested in Redux.
7) Really easy to do. Just install the plugin from the WordPress.org repo and activate the sample config. You'll be able to instantly try out Redux. :)

@everyone   I do wish you luck. I'll gladly give input to help you build your framework. Just if you fork Vafpress, don't let your fork die. Find a way to keep it alive even if you leave. I hate watching all these frameworks come and go. It just seems like a lot of wasted effort. I personally would rather everyone just works together.  ;)

**Comment by scrobbleme** [comment]
@dovy Redux' documentation interface is quite confusing, article-like and overloaded with fancy stuff. Vafpress has an easy-to-navigate-sidebar for instance... It's less a design issue, it's a usability issue.
That makes "getting start" quite hard ...

**Comment by dovy** [comment]
@scrobbleme I wish I could find a way to make it work for you. Vafpress is using bootstrap HTML for docs, while Redux is using WordPress. We also have search... I fear I don't know how to satisfy your needs short of changing to a plain HTML format. :(

**Comment by scrobbleme** [comment]
@dovy Maybe just a structure menu somewhere (on the left) ;)

**Comment by Zackio** [comment]
I looked at a lot of different frameworks, tried them out, all the active ones I find, it came down to this and redux. This was free and I was paying out for a lot already, it all adds up. I'm very happy with vafpress and I want to see it kept alive. I have put up my extra fields I created if any one is interested. So for any one wanting to get together to keep this alive, count me in.

**Comment by padalec** [comment]
Hi, 
there is the way too add a meta box too quick edit form?

**Comment by BassemN** [comment]
I'm using Option-Tree framework and it support Metboxes also It has UI bulider for theme options
https://github.com/valendesigns/option-tree

**Comment by padalec** [comment]
@BassemN there is demo for this framework and? I don't see on-line documentation.

**Comment by BassemN** [comment]
@padalec  There is no demo fro it but there are some screenshots here https://wordpress.org/plugins/option-tree/screenshots/
Also note that OptionTree is a project partly sponsored by ThemeForest, the largest WordPress theme marketplace on the web.

**Comment by padalec** [comment]
@dovy I think about use REDUX to my new theme. Can you explain me how the licens work? I need framework to help me build: theme options, widgets, meta boxes and shortcodes but http://reduxframework.com/extension/metaboxes/?ref=2 meta box plugin costs 130$!! (I it's very expensive) And I can only use in one theme? How this license work with theme developer? Users which will buy my theme should will buy a Redux single license too?

**Comment by scrobbleme** [comment]
@padalec Please ask your Redux questions at the Redux site: http://reduxframework.com

This is for Vafpress and only Vafpress.

And please don't ask any pre-sales question within bug or feature reports. This is just annyoing.

**Comment by padalec** [comment]
Ok sorry

**Comment by rafinkarki** [comment]
Hi i want to add a field in my theme options with radio button options? How is that possible?
